sympy/calculus/util.py

```diff
@@ -253,6 +253,83 @@ def elm_domain(expr, intrvl):
         return _domain
 
 
+def periodicity(f, symbol):
+    """Returns the period of a given function, if possible.
+
+    Examples
+    ========
+    >>> from sympy import Symbol, sin, cos
+    >>> from sympy.calculus.util import periodicity
+    >>> x = Symbol('x')
+    >>> f = sin(x) + sin(2*x) + sin(3*x)
+    >>> periodicity(f, x)
+    2*pi
+    >>> periodicity(sin(x)*cos(x), x)
+    pi
+    """
+    from sympy import simplify, lcm_list
+    from sympy.functions.elementary.trigonometric import TrigonometricFunction
+
+    orig_f = f
+    f = simplify(orig_f)
+    period = None
+
+    if isinstance(f, TrigonometricFunction):
+        period = f.period(symbol)
+
+    if f.is_Mul:
+        _, g = f.as_independent(symbol, as_Add=False)
+        if isinstance(g, TrigonometricFunction):
+            period = g.period(symbol)
+
+    if f.is_Add:
+        k, g = f.as_independent(symbol)
+        if k is not S.Zero:
+            return periodicity(g, symbol)
+
+        periods = []
+        for h in g.args:
+            if h.is_Mul:
+                coeff, _ = h.as_independent(symbol)
+                if coeff.is_number and coeff < 0:
+                    raise NotImplementedError("Periods of only certain "
+                                              "trigonometric functions can be "
+                                              "calculated using this method.")
+
+            periods.append(periodicity(h, symbol))
+
+        period = lcm_fraction(periods)
+
+    if period is not None and orig_f.subs(symbol, symbol + period) == orig_f:
+        return period
+
+    else:
```

Comment:
```
>>> periodicity(x + sin(x), x)
NotImplementedError: Periods of only certain trigonometric functions can be calculated using this method.
```
Can you add few lines about "certain trigonometric functions" for which it works fine currently.